### PR TITLE
Add jax.scipy.signal.detrend

### LINKED ISSUE #3498: scipy.signal.detrend implementation in JAX
HI, 

Would it be possible to implement scipy.signal.detrend in JAX?  I need to compute the gradient of a function which involves de-trending a signal computed from input parameters.

Thanks,
York

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #3498

### CODE REVIEW

**Comment by jakevdp** [comment]
I'm seeing precision issues on GPU & TPU. Will push some fixes.

**Comment by jakevdp** [comment]
GPU & TPU tests are passing now.

**Review by mattjj — APPROVED** [review]
Nice!
